platform/android/jni_utils.cpp

```diff
@@ -395,10 +409,10 @@ Variant::Type get_jni_type(const String &p_type) {
 		idx++;
 	}
 
-	return Variant::NIL;
+	return Variant::OBJECT;
```

Comment:
So this assumes we have an object if we don't match any of the other types, correct?
Should it also checks if the type starts with `[L`?

Reply:
> So this assumes we have an object if we don't match any of the other types, correct?

Yes

> Should it also checks if the type starts with L?

This particular function gets class names (for non-arrays) as plain strings without the `L` prefix and `;` suffix. So, for example, I made a `JNISingleton` in a Godot Android plugin that takes a `java.time.format.DateTimeFormatter` as an argument, and this function gets exactly the string `"java.time.format.DateTimeFormatter"`.